### PR TITLE
Add child store chaining possibility

### LINKED ISSUE #68: Return ElmStore from addChildStore
Return ElmStore from addChildStore inside ElmStore in order to support chaining addChildStore callse

### PR DESCRIPTION
**Pull request by sergei-lapin**

### CODE REVIEW

**Comment by sergei-lapin** [comment]
> It would be convenient to add the same possibility to ElmChildStore from rxjava-2 support module as well. I guess it could be done by moving addChildStore method to the Store interface to make it available for all inheritors. Will you have time to implement this?

Sure

**Comment by sergei-lapin** [comment]
btw now, with the change of base interface, I believe it's worth raising minor version rather then patch for the next release

**Comment by diklimchuk** [comment]
> btw now, with the change of base interface, I believe it's worth raising minor version rather then patch for the next release

I agree, that would be appropriate

File: elmslie-core/src/test/java/vivid/money/elmslie/core/store/ElmStoreWithChildTest.kt [hunk 1]
```diff
@@ -157,11 +164,11 @@ class ElmStoreWithChildTest {
         state: ParentState,
         reducer: StateReducer<ParentEvent, ParentState, ParentEffect, ParentCommand> = NoOpReducer(),
         actor: Actor<ParentCommand, ParentEvent> = NoOpActor()
-    ) = ElmStore(state, reducer, actor)
+    ): Store<ParentEvent, ParentEffect, ParentState> = ElmStore(state, reducer, actor)
```
Comment (diklimchuk) [inline elmslie-core/src/test/java/vivid/money/elmslie/core/store/ElmStoreWithChildTest.kt:167]: Thank you for adding explicit tests for this change

**Pull request merged**